lib/shelley/src/Cardano/Wallet/Shelley/Pools.hs

```diff
@@ -592,27 +615,51 @@ monitorStakePools tr gp nl DBLayer{..} =
 monitorMetadata
     :: Tracer IO StakePoolLog
     -> GenesisParameters
-    -> (   PoolId
-        -> StakePoolMetadataUrl
-        -> StakePoolMetadataHash
-        -> IO (Maybe StakePoolMetadata)
-       )
     -> DBLayer IO
     -> IO ()
-monitorMetadata tr gp fetchMetadata DBLayer{..} = forever $ do
-    refs <- atomically (unfetchedPoolMetadataRefs 100)
-    successes <- fmap catMaybes $ forM refs $ \(pid, url, hash) -> do
-        fetchMetadata pid url hash >>= \case
-            Nothing -> Nothing <$ do
-                atomically $ putFetchAttempt (url, hash)
-
-            Just meta -> Just hash <$ do
-                atomically $ putPoolMetadata hash meta
-
-    when (null refs || null successes) $ do
-        traceWith tr $ MsgFetchTakeBreak blockFrequency
-        threadDelay blockFrequency
+monitorMetadata tr gp DBLayer{..} = do
+    initSettings <- atomically readSettings
+    sref <- newIORef initSettings
+    forever $ do
+        prevSettings <- readIORef sref
+        newSettings <- atomically readSettings
+        writeIORef sref newSettings
+
+        -- delete old metadata table if need be
```

Comment:
Note that this is required to avoid race conditions. We can't drop the table in the PUT logic. Another side effect here is that changing the setting actually takes a bit of time (seconds), because we need to wait for the loop to continue before the new settings get picked up. I feel this is a little awkward @rvl @KtorZ

Reply:
> that changing the setting actually takes a bit of time (seconds), because we need to wait for the loop to continue before the new settings get picked up

I don't quite see the problem with that? So long as: 

- An API call changing the settings is resolved promptly 
- The new settings is saved before the API call resolves

Then of course, the settings will only enter in effects on the next fetching loop. I don't think this is awkward? Why awkward?